libbpf-tools/oomkill.bpf.c

```diff
@@ -1,29 +1,26 @@
 // SPDX-License-Identifier: GPL-2.0
-// Copyright (c) 2022 Jingxiang Zeng
+// Copyright (c) 2022 Jingxiang Zeng and Krisztian Fekete
 #include <vmlinux.h>
 #include <bpf/bpf_helpers.h>
 #include <bpf/bpf_core_read.h>
 #include <bpf/bpf_tracing.h>
-
+#include "compat.bpf.h"
 #include "oomkill.h"
 
-struct {
-	__uint(type, BPF_MAP_TYPE_PERF_EVENT_ARRAY);
-	__uint(key_size, sizeof(u32));
-	__uint(value_size, sizeof(u32));
-} events SEC(".maps");
-
 SEC("kprobe/oom_kill_process")
 int BPF_KPROBE(oom_kill_process, struct oom_control *oc, const char *message)
 {
-	struct data_t data;
+	struct data_t *data;
+	data = reserve_buf(sizeof(*data));
+	if (!data)
+		return 0;
 
-	data.fpid = bpf_get_current_pid_tgid() >> 32;
-	data.tpid = BPF_CORE_READ(oc, chosen, tgid);
-	data.pages = BPF_CORE_READ(oc, totalpages);
-	bpf_get_current_comm(&data.fcomm, sizeof(data.fcomm));
-	bpf_probe_read_kernel(&data.tcomm, sizeof(data.tcomm), BPF_CORE_READ(oc, chosen, comm));
-	bpf_perf_event_output(ctx, &events, BPF_F_CURRENT_CPU, &data, sizeof(data));
+	data->fpid = bpf_get_current_pid_tgid() >> 32;
+	data->tpid = bpf_get_current_pid_tgid();
```

Comment:
This is incorrect, please keep `BPF_CORE_READ(oc, chosen, tgid)`.

Reply:
This is fixed in `3d1a09fb8d2866704f7150af9b12970605160227`.

Reply:
Please rebase and squash your commits properly.

Reply:
Done.

Reply:
Actually, you should not squash into one commit. The orginal changes are good, except the above comment.

Reply:
Is there a value in getting all the individual commits in this case, though? There was no real valuable information/state in the intermediate commits, and it's cleaner now.

Reply:
Hi @chenhengqi, can we have the workflows running to test the PR?

Reply:
Just triggered the CI.

> Is there a value in getting all the individual commits in this case, though? There was no real valuable information/state in the intermediate commits, and it's cleaner now.

Since the PR is meant to change the oomkill tool, I'd expect the change to mountsnoop goes to its own commit.

We accept PR that contains multiple commits.

Reply:
Thank you, done. Now https://github.com/iovisor/bcc/pull/4282/commits/c2931527e5f0198a90878ee5f46daf1e4da76358 has the mountsnoop changes only.

Reply:
One more request, please merge the first two commits. Thanks.

Reply:
Done. Thank you for the review!

Reply:
@chenhengqi, do you need any more changes? I can see the checks have passed as well.